src/Command/PublishOnMarketplaceCommand.php

```diff
@@ -150,15 +156,24 @@ protected function initialize(InputInterface $input, OutputInterface $output): v
             throw new Exception('No API Key is set to authenticate the request to the marketplace. Please set the env var MARKETPLACE_API_KEY or the option --api-key=[...]');
         }
 
-        foreach (['archive', 'changelog', 'metadata-json'] as $option) {
+        foreach (['archive', 'metadata-json'] as $option) {
             if (empty($input->getOption($option))) {
                 throw new Exception(sprintf('Option --%s must be set.', $option));
             }
         }
 
+        if (empty($input->getOption('changelog')) && empty($input->getOption('changelog-file'))) {
+            throw new Exception(sprintf('Either option --changelog or --changelog-file must be set.'));
+        }
+
+        if (!empty($input->getOption('changelog-file'))) {
+            $this->changelog = (new MetadataFile($input->getOption('changelog-file')))->getContent();
```

Comment:
Suggestion: what happens if `changelog-file` is not valid? an invalid path, or a file that cannot be read (bad permission)?